File: java/src/main/java/model/Currency.java
package model;

import external.CurrencyIsoCode;

public enum Currency {
    Euro("€"),
    Dollar("$"),
    Pound("£"),
    Yen("¥");

    Currency(String symbol) {}

    public CurrencyIsoCode toIsoCode() {
        return switch (this) {
            case Euro -> CurrencyIsoCode.EUR;
            case Dollar -> CurrencyIsoCode.USD;
            case Pound -> CurrencyIsoCode.GBP;
            case Yen -> CurrencyIsoCode.JPY;
        };
    }
}

File: kotlin/src/main/kotlin/model/Currency.kt
package model

import external.CurrencyIsoCode

enum class Currency {
    EURO, DOLLAR, POUND, YEN;

    fun toIsoCode(): CurrencyIsoCode {
        return when (this) {
            EURO -> CurrencyIsoCode.EUR
            DOLLAR -> CurrencyIsoCode.USD
            POUND -> CurrencyIsoCode.GBP
            YEN -> CurrencyIsoCode.JPY
        }
    }
}